### PR TITLE
Prohibit keyword usage in autoloads

### LINKED ISSUE #17494: Singletons can have keywords for a name
**Godot version:**
3.0 stable win64

**Issue description:**
In the autoload tab, in project settings you can set singleton name to, for example, "const". You will be unable to use this singleton in gdscript, as editor assumes you are using a keyword.
[MVP.zip](https://github.com/godotengine/godot/files/1808453/MVP.zip)

**Steps to reproduce:**
add a singleton with name that is a reserved keyword in gdscript

### PR DESCRIPTION
**Pull request by Chaosus**
Fix #17494

### CODE REVIEW

File: editor/editor_autoload_settings.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "core/global_constants.h"
 #include "core/project_settings.h"
 #include "editor_node.h"
+#include "modules/gdscript/gdscript.h"
```
Comment (akien-mga) [inline editor/editor_autoload_settings.cpp]: We shouldn't have dependency on modules in `editor/` normally, as you can compile the engine with `module_gdscript_enabled=no`. So at least there should be `#ifdef GDSCRIPT_ENABLED` guards, but ideally this should call to the abstract ScriptLanguage API somehow.

Comment (Chaosus) [reply editor/editor_autoload_settings.cpp]: Ah.. didnt knew that its possible and used somehow. GDScript is such a core engine element so its hardly to even present in mind that Godot can be used without it.

**Comment by vnen** [comment]
I don't know if having a GDScript dependency inside the editor code is a good idea.

**Comment by mhilbrunner** [comment]
So, Akiens suggestion was, if I'm right:

1. Check `ScriptServer.is_scripting_enabled()`.
1. If scripting is enabled, check how many languages are registered (`ScriptServer.get_language_count()`)
1. Iterate over each `ScriptLanguage` using that count with `ScriptServer.get_language(i)`
1. For each returned `ScriptLanguage`, retrieve `ScriptLanguage.get_reserved_words(*list)`
1. Build a set out of those lists, use that to check against the autoload names

That would only be a dependency on `core/script_language.h`, no modules.

IMO, this is better than harcoding this list in yet another place, but we could also easily merge this and I'll try to do the above in a later cleanup PR. :) It *is* aleady hardcoded in multiple places anyway, AFAIK.

**Comment by Chaosus** [comment]
Alright, that sounds hard.. I wash my hands and close this PR :)

**Pull request closed without merging**